### ISSUE TITLE
object as tags (and text too)

### ISSUE THREAD

**Issue by alfredopacino** [bootstrap-tagsinput/bootstrap-tagsinput#461]
Hi, I have to make a mixed field:
- objects taken from the db
- free input
this is my config
```
    var users = new Bloodhound({
        datumTokenizer: function(d) {
            return Bloodhound.tokenizers.whitespace(d.name);
        },
        queryTokenizer: Bloodhound.tokenizers.whitespace,
        remote: {
            url: '/api/community/users?q=%QUERY',
            wildcard: '%QUERY'
        }
    });
    users.initialize();

    $('#tagsinput').tagsinput({
        typeaheadjs: {
            name: 'name',
            displayKey: 'name',
            itemValue: '_id',
            valueKey: '_id',
            source: users.ttAdapter()
        }
    });
```

my json response from /api/community/users
```
[{
	"_id": "56b351e01680a9c815efbaa8",
	"name": "John",
	"age": 20,
	"__v": 0
}]
```

So I'm selecting users, from the db and creating new users as free text.
**The problem is**: when I call $('#tagsinput').tagsinput('items')
I'm expecting something like:
```
[{
	"_id": "56b351e01680a9c815efbaa8",
	"name": "John",
	"age": 20,
	"__v": 0
},{
name:"New user"
}]
```

instead of an array of strings.
I'm sure I messed up typeahead config when I init tagsinput.

PS: why in documentation tagsinput docs I see displayKey and valueKey, isn't this an old typehead usage?

**Comment by alfredopacino** [comment]
Watching the issue section I guess this project is dead, am I right?

**Comment by lucknerjb** [comment]
@alfredopacino Nope, not dead. Have had a bit of trouble keeping up to date but I'm getting back on it :)

The typeahead implementation needs some major updates. I'm planning on making that a priority in the near future.

**Comment by chubz** [comment]
is there any news on this?
It seems like a logical feature: If you connect typeahead and have objects as tags to have porpper backend behavior then it should stil be possible to freeInput custom tags.

**Comment by antonioaltamura** [comment]
@chubz for this specific project a year ago I switched to Angularjs (v1) for various reason and this problem solved itself.

**Comment by chubz** [comment]
Yeah, really bad idea to involve a complete framework into my project to enable this one feature. Also being dependant on angular for it is bad idea in general. I hope there is a release coming for jQuery version, that is all. I'm going to try to patch it but I'm really inexperienced in JS so don't know if it will be of any value and if I actually do it.

**Comment by antonioaltamura** [comment]
Obiouvlsy invole Angular for this feature only is non-sense. It can be helpful if you have several complex form though (keeping in mind you have to completely rewrite your application)
